cmd/internal/cli/cache_list_linux.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cli
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/sylabs/singularity/docs"
+	"github.com/sylabs/singularity/internal/app/singularity"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+var (
+	typeNameList string
+	allList      bool
+)
+
+func init() {
+	CacheListCmd.Flags().SetInterspersed(false)
+
+	CacheListCmd.Flags().StringVarP(&typeNameList, "type", "T", "", "list cache type, choose between: library, and oci")
```

Comment:
`CacheListCmd.Flags().StringSliceVarP(&cacheListTypes, "type", "T", []string{"library", "oci"}, "list cache type, choose between: library, oci, and blob")`

There are two things that we're doing here which improve this code:

1. We're using `pflag` to convert whatever argument was passed to `--type <...>` and convert it into a `[]string`. It's a convenient way to avoid parsing the arguments to a flag
2. We're using `pflag` to set a default value to the flag. This way, we don't have to manually check that the `len(typeNameList) > 0`, we simply just assume that it's been set. If the user *doesn't* specify `--type <...>`, then `cacheListTypes` will be set to `[]string{"library", "oci"}`. Again, this is both a convenience thing but also it will clean up unnecessary conditionals from our code